files/class-api-client.php

```diff
@@ -152,13 +152,20 @@ private function calculate_upload_timeout( $file_size ) {
 
 	public function get_file( $file_path ) {
 		// check in cache first
-		$file_content = $this->cache->get_file( $file_path );
-		if ( $file_content ) {
-			return $file_content;
+		$file = $this->cache->get_file( $file_path );
+		if ( $file ) {
+			return $file;
 		}
 
+		$tmp_file = tempnam( get_temp_dir(), 'vip' );
```

Comment:
What's the advantage to moving this cache file creation out of the cache class and into the api client class? It seems to me better to encapsulate all the details inside the cache class.

Reply:
This was mainly a side-effect of having to pass the file as the arguments for `call_api()`. I'm not sure if it's better encapsulated in the cache class though. It _should_ be encapsulated but I think the API client class works just as well